src/main/Config.cpp

```diff
@@ -286,6 +286,117 @@ Config::addHistoryArchive(std::string const& name, std::string const& get,
     }
 }
 
+static std::array<std::string, 3> const kQualities = {"LOW", "MEDIUM", "HIGH"};
+
+std::string
+Config::toString(ValidatorQuality q)
+{
+    return kQualities[static_cast<int>(q)];
+}
+
+Config::ValidatorQuality
+Config::parseQuality(std::string const& q)
+{
+    auto it = std::find_if(kQualities.begin(), kQualities.end(),
+                           [&](std::string const& v) { return (v == q); });
+
+    ValidatorQuality res;
+
+    if (it != kQualities.end())
+    {
+        res = static_cast<Config::ValidatorQuality>(
+            std::distance(kQualities.begin(), it));
+    }
+    else
+    {
+        throw std::invalid_argument(fmt::format("Unknown QUALITY {}", q));
+    }
+    return res;
+}
+
+std::vector<Config::ValidatorEntry>
+Config::parseValidators(
+    std::shared_ptr<cpptoml::base> validators)
+{
+    std::vector<ValidatorEntry> res;
+
+    auto tarr = validators->as_table_array();
+    if (!tarr)
+    {
+        throw std::invalid_argument("malformed VALIDATORS");
+    }
+    for (auto const& valRaw : *tarr)
+    {
+        auto validator = valRaw->as_table();
+        if (!validator)
+        {
+            throw std::invalid_argument("malformed VALIDATORS");
+        }
+        ValidatorEntry ve;
+        std::string pubKey, hist;
+        bool qualitySet = false;
+        for (auto const& f : *validator)
+        {
+            if (f.first == "NAME")
+            {
+                ve.mName = readString(f);
+            }
+            else if (f.first == "HOME_DOMAIN")
+            {
+                ve.mHomeDomain = readString(f);
+            }
+            else if (f.first == "QUALITY")
+            {
+                auto q = readString(f);
+                ve.mQuality = parseQuality(q);
+                qualitySet = true;
+            }
+            else if (f.first == "PUBLIC_KEY")
+            {
+                pubKey = readString(f);
+            }
+            else if (f.first == "ADDRESS")
+            {
+                auto address = readString(f);
+                KNOWN_PEERS.emplace_back(address);
+            }
+            else if (f.first == "HISTORY")
+            {
+                hist = readString(f);
+            }
+            else
+            {
+                throw std::invalid_argument("malformed VALIDATORS entry");
+            }
+        }
+        if (ve.mName.empty() || pubKey.empty() || ve.mHomeDomain.empty())
+        {
+            throw std::invalid_argument(
+                fmt::format("malformed VALIDATORS entry {}", ve.mName));
```

Comment:
when condition ve.mName.empty() is true this message will not be as useful as it could be

Reply:
fixed